app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -84,6 +85,10 @@ private GsonDatabaseAdapter(final IServiceLocator serviceLocator) {
     @Override
     public void addContact(final Contact contact) throws IOException {
         // Get the list of contacts and add the new contact
+        if(getContactWithName(contact.getName()) != null) {
```

Comment:
I'd appreciate if this was it's own , standradized, method in the db as I had to implement one like this myself but differently.